api/lib/domain/services/check-sco-account-recovery-service.js

```diff
@@ -19,6 +24,12 @@ async function retrieveSchoolingRegistration({
     userReconciliationService,
   });
 
+  const accountRecoveryDemands = await accountRecoveryDemandRepository.findByUserId(userId);
+
+  if (accountRecoveryDemands.some((accountRecoveryDemand) => accountRecoveryDemand.used)) {
+    throw new UserHasAlreadyLeftSCO();
+  }
```

Comment:
Question: Since this is done in 3 places, would it be worth exporting it into a function ?